In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../Load_data/DataSet_WT.csv")

In [3]:
# en este podemos juntar hp, y borrar dmg_taken

In [4]:
df.head()

character  max_hp  entering_hp  damage_taken  enemies  level_asc_floor  \
0   WATCHER      72           72             0       40                3   
1   WATCHER      72           72             0        7                5   
2   WATCHER      72           72            15       55                8   
3   WATCHER      72           57            17       34               12   
4   WATCHER      72           61             0        3               19   

   dmg_mean_ascension enemy_type  Adaptation  Adaptation+1  ...  Turnip  \
0            8.968972      comun           0             0  ...       0   
1            8.968972      comun           0             0  ...       0   
2            8.968972      comun           0             0  ...       0   
3            8.968972      comun           0             0  ...       0   
4            8.968972      comun           0             0  ...       0   

   Unceasing Top  Vajra  Velvet Choker  VioletLotus  WarpedTongs  Whetstone  \
0              0      0              0            0            0          0   
1              0      0              0            0            0          0   
2              0      0              0            0            0          0   
3              0      0              0            0            0          0   
4              0      0              0            0            0          0   

   White Beast Statue  WingedGreaves  Yang  
0                   0              0     0  
1                   0              0     0  
2                   0              0     0  
3                   0              0     0  
4                   1              0     0  

[5 rows x 390 columns]

In [5]:
df.level_asc_floor.value_counts()

level_asc_floor
7       346
8       310
3       307
5       269
25      266
       ... 
1750      1
980       1
1778      1
1766      1
1889      1
Name: count, Length: 813, dtype: int64

In [7]:
df.columns

Index(['character', 'max_hp', 'entering_hp', 'damage_taken', 'enemies',
       'level_asc_floor', 'dmg_mean_ascension', 'enemy_type', 'Adaptation',
       'Adaptation+1',
       ...
       'Turnip', 'Unceasing Top', 'Vajra', 'Velvet Choker', 'VioletLotus',
       'WarpedTongs', 'Whetstone', 'White Beast Statue', 'WingedGreaves',
       'Yang'],
      dtype='object', length=390)

In [8]:
df.drop(columns=["character", "damage_taken", "enemies", "enemy_type", "dmg_mean_ascension"], inplace=True)

In [ ]:
import pandas as pd
from sklearn.ensemble import ExtraTreesRegressor
from sklearn.preprocessing import OneHotEncoder, MaxAbsScaler
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score

# Suponiendo que tu DataFrame se llama df
# df = pd.read_csv('your_dataset.csv') # Carga tu dataset aquí

# Separar características y etiqueta
X = df.drop(columns=["level_asc_floor"])
y = df["level_asc_floor"]

# Dividir en conjunto de entrenamiento y prueba
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Estandarizar los datos usando MaxAbsScaler
scaler = MaxAbsScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Entrenar el modelo Extra Trees Regressor
model = ExtraTreesRegressor(n_estimators=100, random_state=42)
model.fit(X_train_scaled, y_train)

# Realizar predicciones
y_pred = model.predict(X_test_scaled)

# Calcular R^2
r2 = r2_score(y_test, y_pred)

print(f"R^2 Score: {r2}")

# Otras métricas opcionales
from sklearn.metrics import mean_absolute_error, mean_squared_error
mae = mean_absolute_error(y_test, y_pred)
mse = mean_squared_error(y_test, y_pred)

print(f"Mean Absolute Error: {mae}")
print(f"Mean Squared Error: {mse}")

R^2 Score: 0.9774695189111515
Mean Absolute Error: 45.86361547207637
Mean Squared Error: 10035.370716365062


Normal: 

R^2 Score: 0.976961108564015    Mean Absolute Error: 48.07894334485287  Mean Squared Error: 10261.823329135825

Drop daño recibido:

R^2 Score: 0.9771165249806509   Mean Absolute Error: 47.38598062624359  Mean Squared Error: 10192.59882610811

Drop enemies, daño recibido: 

R^2 Score: 0.9774695189111515   Mean Absolute Error: 45.86361547207637  Mean Squared Error: 10035.370716365062
